specs/TrackingPrevention.md

```diff
@@ -0,0 +1,205 @@
+Tracking Prevention
+===
+
+# Background
+The WebView2 team has been asked for an API to toggle tracking prevention and also ability to control levels
+of tracking prevention.
+
+We are proposing two API's
+IsTrackingPreventionEnabled: This API allows you to enable/disable the tracking prevention feature. By default
+tracking prevention feature is enabled and set to `Balanced` level for WebView2. You can set the
+`CoreWebView2EnvironmentOptions.IsTrackingPreventionEnabled` property to false to disable the tracking prevention
+feature for WebView2 before creating environment that also skips the related code and improves the performance.
+
+TrackingPreventionLevel: This API allows you to control levels of tracking prevention for WebView2 which
+are associated with a profile when the feature is enabled and otherwise TrackingPreventionLevel is
+CoreWebView2TrackingPreventionLevel.Off and can't be changed to other kinds.
+The levels are similar to Edge: `Off`, `Basic`, `Balanced` and `Strict`.
+
+For reference, in the screenshot below, this API sets the levels of tracking prevention as a WebView2 API.
+
+![Edge Tracking Prevention](images/TrackingPreventionLevels.png)
+
+# Examples
+## IsTrackingPreventionEnabled
+
+```c#
+Create WebView Environment with option that disable tracking prevention feature.
+
+void CreateEnvironmentWithOption()
+{
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    // If we're displaying app content or otherwise are not worried about tracking, we can
+    // disable the tracking prevention feature to improve runtime performance.
+    options.IsTrackingPreventionEnabled = false;
+    CoreWebView2Environment environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+}
+```
+
+```cpp
+Microsoft::WRL::ComPtr<ICoreWebView2EnvironmentOptions3> options3;
+if (options.As(&options3) == S_OK)
+{
+    // If we're displaying app content or otherwise are not worried about tracking, we can
+    // disable the tracking prevention feature to improve runtime performance.
+    CHECK_FAILURE(options3->put_IsTrackingPreventionEnabled(FALSE));
+}
+```
+
+## TrackingPreventionLevel
+Example of code in end user facing UI to change the TrackingPreventionLevel.
+
+```c#
+
+void SetTrackingPreventionLevel(CoreWebView2TrackingPreventionLevel value)
+{
+    WebViewProfile.TrackingPreventionLevel = value;
+    MessageBox.Show(this, "Tracking prevention level is set successfully", "Tracking Prevention Level");

```

Comment:
Currently, it doesn't do anything if IsTrackingPreventionEnabled is false, and we documented accordingly. However, we wanted to bring this in meeting either to throw an error or allow and do nothing.

Reply:
We could call it PreferredTrackingPreventionLevel to make it clearer that this may not be honored.

Reply:
See above.

Reply:
And now let's discuss changing the property name: 
* PreferredTrackingPreventionLevel 

But leave the enum name and value names as is.

Add 'Preferred' as a prefix since it may not be used if the other property disables the whole feature.